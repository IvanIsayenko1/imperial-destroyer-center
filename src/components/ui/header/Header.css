header {
  color: white;
  padding: 1rem;
  margin-bottom: 2rem;

  border-bottom: 1px solid var(--accent-color);
}

.header-wrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 1280px;
  margin: 0 auto;
  gap: 2rem;
}

.menu-button {
  display: none;
  background: none;
  border: none;
  color: var(--rebel-base);
  font-size: 1.5rem;
  cursor: pointer;
  padding: 0.5rem;
}

@media (max-width: 768px) {
  .header-wrapper {
    flex-direction: column;
    align-items: stretch;
    gap: 1rem;
  }

  .header-top {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .menu-button {
    display: block;
  }

  .navigation-container {
    display: none;
  }

  .navigation-container.open {
    display: block;
  }

  nav {
    flex-direction: column;
  }
}
