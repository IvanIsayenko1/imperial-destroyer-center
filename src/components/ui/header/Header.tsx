import React, { useState } from 'react';
import './Header.css';
import { Navigation } from '../navigation/Navigation';
import { Title } from '../title/Title';

interface HeaderProps {
  title?: string;
}

export const Header: React.FC<HeaderProps> = ({
  title = 'Imperial Destroyer Center',
}) => {
  const [isMenuOpen, setIsMenuOpen] = useState(false);

  return (
    <header>
      <div className="header-wrapper">
        <div className="header-top">
          <Title title={title} />
          <button
            className="menu-button"
            onClick={() => setIsMenuOpen(!isMenuOpen)}
            aria-label="Toggle navigation"
          >
            â˜°
          </button>
        </div>
        <div className={`navigation-container ${isMenuOpen ? 'open' : ''}`}>
          <Navigation />
        </div>
      </div>
    </header>
  );
};
