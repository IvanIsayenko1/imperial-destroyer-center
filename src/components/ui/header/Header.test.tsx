import { render, screen, fireEvent } from "@testing-library/react";
import { Header } from "./Header";

vi.mock("../navigation/Navigation", () => ({
  Navigation: () => <div>Navigation</div>,
}));

vi.mock("../title/Title", () => ({
  Title: ({ title }: { title: string }) => <div>{title}</div>,
}));

describe("Header component", () => {
  it("should render the title with default value", () => {
    render(<Header />);
    expect(screen.getByText("Imperial Destroyer Center")).toBeInTheDocument();
  });

  it("should render the title with provided title prop", () => {
    const customTitle = "Custom Title";
    render(<Header title={customTitle} />);
    expect(screen.getByText(customTitle)).toBeInTheDocument();
  });

  it("should toggle the menu on button click", () => {
    render(<Header />);
    const menuButton = screen.getByLabelText("Toggle navigation");
    const navigationContainer = screen.getByText("Navigation").parentElement;

    expect(navigationContainer).not.toHaveClass("open");

    fireEvent.click(menuButton);

    expect(navigationContainer).toHaveClass("open");

    fireEvent.click(menuButton);

    expect(navigationContainer).not.toHaveClass("open");
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
