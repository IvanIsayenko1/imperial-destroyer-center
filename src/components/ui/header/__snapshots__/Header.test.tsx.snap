// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <div
      class="header-wrapper"
    >
      <div
        class="header-top"
      >
        <div>
          Imperial Destroyer Center
        </div>
        <button
          aria-label="Toggle navigation"
          class="menu-button"
        >
          â˜°
        </button>
      </div>
      <div
        class="navigation-container "
      >
        <div>
          Navigation
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
